packages/visualizations/src/components/Map/Choropleth.svelte

```diff
@@ -2,58 +2,60 @@
 
 <script lang="ts">
     import turfBbox from '@turf/bbox';
+    import { debounce } from 'lodash';
     import type { SourceSpecification } from 'maplibre-gl';
-    // eslint-disable-next-line import/no-unresolved
-    import type { BBox, FeatureCollection } from 'geojson';
-    import type { ColorsScale, DataBounds, Color } from '../types';
+    import type { BBox } from 'geojson';
+    import type { ColorScales, DataBounds, Color } from '../types';
     import MapRender from './MapRender.svelte';
     import { BLANK } from './mapStyles';
-    import { colorShapes, LIGHT_GREY, DARK_GREY } from './utils';
+    import { getDataBounds, mapKeyToColor, isVectorTile, VOID_BOUNDS } from './utils';
+    import { DEFAULT_COLORS, DEFAULT_COLORS_SCALE } from './constants';
+    import { ChoroplethShapeTypes } from './types';
     import type {
         ChoroplethDataValue,
         ChoroplethLayer,
         ChoroplethOptions,
         MapRenderTooltipFunction,
-        ChoroplethShapeValue,
+        ChoroplethShapeValues,
         ChoroplethTooltipFormatter,
         MapLegend,
     } from './types';
 
     export let data: { value: ChoroplethDataValue[] }; // values, and the key to match
     export let options: ChoroplethOptions; // contains the shapes to display & match
 
-    let shapes: ChoroplethShapeValue;
-    let colorsScale: ColorsScale;
-
-    const defaultColorsScale: ColorsScale = {
-        type: 'gradient',
-        colors: {
-            start: LIGHT_GREY,
-            end: DARK_GREY,
-        },
-    };
-
-    const defaultEmptyValueColor = '#cccccc';
+    let shapes: ChoroplethShapeValues;
+    let colorsScale: ColorScales;
 
     let aspectRatio: number;
     let renderTooltip: MapRenderTooltipFunction;
-    let bbox: BBox;
+    let bbox: BBox | undefined;
+    let fixedBbox: BBox | undefined;
```

Comment:
Having both `bbox` and `fixedBbox` seems too complicated IMO.
I'm not sure what `bbox` is used for now.

Can't we just have `bbox`, and if it's set, use it, else compute one if we can (= geojson)?